Test Failure: SwitchFromSingleInstanceToFailOverTest.testNewFailOverEnabledInstance FFDCs

Test Failure: com.ibm.ws.concurrent.persistent.fat.failover1serv.SwitchFromSingleInstanceToFailOverTest.testNewFailOverEnabledInstance

```
testNewFailOverEnabledInstance:junit.framework.AssertionFailedError: 2021-01-13-08:30:10:458 [Unexpected FFDC reporting javax.resource.ResourceException was found (count = 12): /WebSphere/Misc/jazz_build/jbe_flash170/jazz/buildsystem/buildengine/eclipse/build/dev/image/output/wlp/usr/servers/com.ibm.ws.concurrent.persistent.fat.failover1serv/logs/ffdc/ffdc_21.01.13_08.29.45.2.log
>------Start of DE processing------ = [1/13/21 8:29:45:855 GMT]
>Exception = javax.resource.ResourceException
>Source = com.ibm.ejs.j2c.ConnectionManager.initializeForUOW
>probeid = 730
>Stack Dump = javax.resource.ResourceException: addSync: caught Exception
> at com.ibm.ejs.j2c.XATransactionWrapper.addSync(XATransactionWrapper.java:180)
> at com.ibm.ejs.j2c.ConnectionManager.initializeForUOW(ConnectionManager.java:973)
> at com.ibm.ejs.j2c.ConnectionManager.involveMCInTran(ConnectionManager.java:692)
> at com.ibm.ejs.j2c.ConnectionManager.allocateConnection(ConnectionManager.java:308)
> at com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection(WSJdbcDataSource.java:137)
> at com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection(WSJdbcDataSource.java:111)
> at org.eclipse.persistence.sessions.JNDIConnector.connect(JNDIConnector.java:135)
> at org.eclipse.persistence.sessions.DatasourceLogin.connectToDatasource(DatasourceLogin.java:162)
> at org.eclipse.persistence.internal.databaseaccess.DatasourceAccessor.connectInternal(DatasourceAccessor.java:346)
> at org.eclipse.persistence.internal.databaseaccess.DatabaseAccessor.connectInternal(DatabaseAccessor.java:309)
> at org.eclipse.persistence.internal.databaseaccess.DatasourceAccessor.reconnect(DatasourceAccessor.java:583)
> at org.eclipse.persistence.internal.databaseaccess.DatabaseAccessor.reconnect(DatabaseAccessor.java:1634)
> at org.eclipse.persistence.internal.databaseaccess.DatasourceAccessor.incrementCallCount(DatasourceAccessor.java:321)
> at org.eclipse.persistence.internal.databaseaccess.DatabaseAccessor.basicExecuteCall(DatabaseAccessor.java:615)
> at org.eclipse.persistence.internal.databaseaccess.DatabaseAccessor.executeCall(DatabaseAccessor.java:560)
> at org.eclipse.persistence.internal.sessions.AbstractSession.basicExecuteCall(AbstractSession.java:2064)
> at org.eclipse.persistence.sessions.server.ClientSession.executeCall(ClientSession.java:310)
> at org.eclipse.persistence.internal.queries.DatasourceCallQueryMechanism.executeCall(DatasourceCallQueryMechanism.java:262)
> at org.eclipse.persistence.internal.queries.DatasourceCallQueryMechanism.executeCall(DatasourceCallQueryMechanism.java:248)
> at org.eclipse.persistence.internal.queries.DatasourceCallQueryMechanism.insertObject(DatasourceCallQueryMechanism.java:398)
> at org.eclipse.persistence.internal.queries.StatementQueryMechanism.insertObject(StatementQueryMechanism.java:165)
> at org.eclipse.persistence.internal.queries.StatementQueryMechanism.insertObject(StatementQueryMechanism.java:180)
> at org.eclipse.persistence.internal.queries.DatabaseQueryMechanism.insertObjectForWrite(DatabaseQueryMechanism.java:489)
> at org.eclipse.persistence.queries.InsertObjectQuery.executeCommit(InsertObjectQuery.java:80)
> at org.eclipse.persistence.queries.InsertObjectQuery.executeCommitWithChangeSet(InsertObjectQuery.java:90)
> at org.eclipse.persistence.internal.queries.DatabaseQueryMechanism.executeWriteWithChangeSet(DatabaseQueryMechanism.java:301)
> at org.eclipse.persistence.queries.WriteObjectQuery.executeDatabaseQuery(WriteObjectQuery.java:58)
> at org.eclipse.persistence.queries.DatabaseQuery.execute(DatabaseQuery.java:908)
> at org.eclipse.persistence.queries.DatabaseQuery.executeInUnitOfWork(DatabaseQuery.java:807)
> at org.eclipse.persistence.queries.ObjectLevelModifyQuery.executeInUnitOfWorkObjectLevelModifyQuery(ObjectLevelModifyQuery.java:108)
> at org.eclipse.persistence.queries.ObjectLevelModifyQuery.executeInUnitOfWork(ObjectLevelModifyQuery.java:85)
> at org.eclipse.persistence.internal.sessions.UnitOfWorkImpl.internalExecuteQuery(UnitOfWorkImpl.java:2900)
> at org.eclipse.persistence.internal.sessions.AbstractSession.executeQuery(AbstractSession.java:1866)
> at org.eclipse.persistence.internal.sessions.AbstractSession.executeQuery(AbstractSession.java:1848)
> at org.eclipse.persistence.internal.sessions.AbstractSession.executeQuery(AbstractSession.java:1799)
> at org.eclipse.persistence.internal.sessions.CommitManager.commitNewObjectsForClassWithChangeSet(CommitManager.java:227)
> at org.eclipse.persistence.internal.sessions.CommitManager.commitAllObjectsWithChangeSet(CommitManager.java:126)
> at org.eclipse.persistence.internal.sessions.AbstractSession.writeAllObjectsWithChangeSet(AbstractSession.java:4286)
> at org.eclipse.persistence.internal.sessions.UnitOfWorkImpl.commitToDatabase(UnitOfWorkImpl.java:1445)
> at org.eclipse.persistence.internal.sessions.UnitOfWorkImpl.commitToDatabaseWithPreBuiltChangeSet(UnitOfWorkImpl.java:1591)
> at org.eclipse.persistence.internal.sessions.RepeatableWriteUnitOfWork.writeChanges(RepeatableWriteUnitOfWork.java:455)
> at org.eclipse.persistence.internal.jpa.EntityManagerImpl.flush(EntityManagerImpl.java:878)
> at com.ibm.ws.concurrent.persistent.db.DatabaseTaskStore.createProperty(DatabaseTaskStore.java:292)
> at com.ibm.ws.concurrent.persistent.internal.InvokerTask.run(InvokerTask.java:299)
> at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:522)
> at java.util.concurrent.FutureTask.run(FutureTask.java:277)
> at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:239)
> at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160)
> at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
> at java.lang.Thread.run(Thread.java:812)
>Caused by: javax.transaction.RollbackException: Transaction rolled back
> at com.ibm.tx.jta.impl.TransactionImpl.registerSynchronization(TransactionImpl.java:2284)
> at com.ibm.tx.jta.embeddable.impl.EmbeddableTranManagerSet.registerSynchronization(EmbeddableTranManagerSet.java:229)
> at com.ibm.tx.jta.embeddable.impl.EmbeddableTranManagerSet.registerSynchronization(EmbeddableTranManagerSet.java:216)
> at com.ibm.ws.transaction.services.TransactionManagerService.registerSynchronization(TransactionManagerService.java:406)
> at com.ibm.ejs.j2c.XATransactionWrapper.addSync(XATransactionWrapper.java:170)
, Unexpected FFDC reporting javax.persistence.PersistenceException was found (count = 2): /WebSphere/Misc/jazz_build/jbe_flash170/jazz/buildsystem/buildengine/eclipse/build/dev/image/output/wlp/usr/servers/com.ibm.ws.concurrent.persistent.fat.failover1serv/logs/ffdc/ffdc_21.01.13_08.30.01.0.log
>------Start of DE processing------ = [1/13/21 8:30:01:168 GMT]
>Exception = javax.persistence.PersistenceException
>Source = com.ibm.ws.concurrent.persistent.db.DatabaseTaskStore
>probeid = 309
>Stack Dump = javax.persistence.PersistenceException: Exception [EclipseLink-4002] (Eclipse Persistence Services - 2.6.8.WAS-v20201111-bb702a8): org.eclipse.persistence.exceptions.DatabaseException
>Internal Exception: java.sql.SQLException: addSync: caught Exception
>Error Code: 0
>Call: INSERT INTO WLPPROP (ID, VAL) VALUES (?, ?)
> bind => [2 parameters bound]
>Query: InsertObjectQuery(com.ibm.ws.concurrent.persistent.db.Property@c18083a0)
> at org.eclipse.persistence.internal.jpa.EntityManagerImpl.flush(EntityManagerImpl.java:883)
> at com.ibm.ws.concurrent.persistent.db.DatabaseTaskStore.createProperty(DatabaseTaskStore.java:292)
> at com.ibm.ws.concurrent.persistent.internal.InvokerTask.run(InvokerTask.java:299)
> at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:522)
> at java.util.concurrent.FutureTask.run(FutureTask.java:277)
> at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:239)
> at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160)
> at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
> at java.lang.Thread.run(Thread.java:812)
>Caused by: Exception [EclipseLink-4002] (Eclipse Persistence Services - 2.6.8.WAS-v20201111-bb702a8): org.eclipse.persistence.exceptions.DatabaseException
>Internal Exception: java.sql.SQLException: addSync: caught Exception
>Error Code: 0
>Call: INSERT INTO WLPPROP (ID, VAL) VALUES (?, ?)
> bind => [2 parameters bound]
>Query: InsertObjectQuery(com.ibm.ws.concurrent.persistent.db.Property@c18083a0)
> at org.eclipse.persistence.exceptions.DatabaseException.sqlException(DatabaseException.java:316)
> at org.eclipse.persistence.sessions.JNDIConnector.connect(JNDIConnector.java:147)
> at org.eclipse.persistence.sessions.DatasourceLogin.connectToDatasource(DatasourceLogin.java:162)
> at org.eclipse.persistence.internal.databaseaccess.DatasourceAccessor.connectInternal(DatasourceAccessor.java:346)
> at org.eclipse.persistence.internal.databaseaccess.DatabaseAccessor.connectInternal(DatabaseAccessor.java:309)
> at org.eclipse.persistence.internal.databaseaccess.DatasourceAccessor.reconnect(DatasourceAccessor.java:583)
> at org.eclipse.persistence.internal.databaseaccess.DatabaseAccessor.reconnect(DatabaseAccessor.java:1634)
> at org.eclipse.persistence.internal.databaseaccess.DatasourceAccessor.incrementCallCount(DatasourceAccessor.java:321)
> at org.eclipse.persistence.internal.databaseaccess.DatabaseAccessor.basicExecuteCall(DatabaseAccessor.java:615)
> at org.eclipse.persistence.internal.databaseaccess.DatabaseAccessor.executeCall(DatabaseAccessor.java:560)
> at org.eclipse.persistence.internal.sessions.AbstractSession.basicExecuteCall(AbstractSession.java:2064)
> at org.eclipse.persistence.sessions.server.ClientSession.executeCall(ClientSession.java:310)
> at org.eclipse.persistence.internal.queries.DatasourceCallQueryMechanism.executeCall(DatasourceCallQueryMechanism.java:262)
> at org.eclipse.persistence.internal.queries.DatasourceCallQueryMechanism.executeCall(DatasourceCallQueryMechanism.java:248)
> at org.eclipse.per
```

This is due to the combination of test infrastructure slowness with a test case that sets a very small value for the missed task threshold.  Tests use the small value in order to run faster, but it makes them vulnerable to transaction timeouts.  The transaction timeouts don't otherwise interfere with the test results because the failover feature that is being tested includes retries, but FFDC messages do get logged and the test needs to tolerate those.

--- a/dev/com.ibm.ws.concurrent.persistent_fat_failover1serv/fat/src/com/ibm/ws/concurrent/persistent/fat/failover1serv/SwitchFromSingleInstanceToFailOverTest.java
+++ b/dev/com.ibm.ws.concurrent.persistent_fat_failover1serv/fat/src/com/ibm/ws/concurrent/persistent/fat/failover1serv/SwitchFromSingleInstanceToFailOverTest.java
@@ -38,6 +38,16 @@ import failover1serv.web.Failover1ServerTestServlet;
  * enable fail over. There are a variety of ways they might do this, which are described in more detail
  * in the comments for each individual test.
  */
+// Various FFDCs due to transaction timeouts from the combination of infrastructure slowness
+// and tests using a very small missedTaskThreshold:
+@AllowedFFDC({
+    "java.lang.IllegalStateException",
+    "javax.persistence.PersistenceException",
+    "javax.resource.ResourceException",
+    "javax.transaction.RollbackException",
+    "javax.transaction.xa.XAException",
+    "org.apache.derby.client.am.XaException"
+})
 @RunWith(FATRunner.class)
 public class SwitchFromSingleInstanceToFailOverTest extends FATServletClient {
 	private static final String APP_NAME = "failover1servApp";
@@ -75,7 +85,6 @@ public class SwitchFromSingleInstanceToFailOverTest extends FATServletClient {
      * testEnableFailOverWhileServerIsRunning - Schedules tasks on an instance where fail over is not enabled.
      * Enables fail over while the server is still running. Then verifies that the previous, as well as new, tasks run.
      */
-    @AllowedFFDC("org.apache.derby.client.am.XaException") // due to roll back & abort due to transaction timeout
     @Test
     public void testEnableFailOverWhileServerIsRunning() throws Exception {
         ServerConfiguration config = originalConfig.clone();
@@ -179,7 +188,6 @@ public class SwitchFromSingleInstanceToFailOverTest extends FATServletClient {
      * testEnableFailOverWhileServerIsStopped - Schedules tasks on an instance where fail over is not enabled.
      * Stops the server and enables fail over. Starts up the server and verifies that the previous, as well as new, tasks run.
      */
-    @AllowedFFDC("org.apache.derby.client.am.XaException") // due to roll back & abort due to transaction timeout
     @Test
     public void testEnableFailOverWhileServerIsStopped() throws Exception {
         ServerConfiguration config = originalConfig.clone();
@@ -289,7 +297,6 @@ public class SwitchFromSingleInstanceToFailOverTest extends FATServletClient {
      * This is not a recommended way of enabling fail over, because it leaves instances with and without fail over running at
      * the same time. But it is being tested here in case anyone tries it.
      */
-    @AllowedFFDC("org.apache.derby.client.am.XaException") // due to roll back & abort due to transaction timeout
     @Test
     public void testNewFailOverEnabledInstance() throws Exception {
         ServerConfiguration config = originalConfig.clone();
@@ -414,7 +421,6 @@ public class SwitchFromSingleInstanceToFailOverTest extends FATServletClient {
      * This is not a recommended way of enabling fail over, because it leaves instances with and without fail over running at
      * the same time. But it is being tested here in case anyone tries it.
      */
-    @AllowedFFDC("org.apache.derby.client.am.XaException") // due to roll back & abort due to transaction timeout
     @Test
     public void testNewFailOverEnabledInstanceWhileServerIsRunning() throws Exception {
         ServerConfiguration config = originalConfig.clone();
@@ -533,7 +539,6 @@ public class SwitchFromSingleInstanceToFailOverTest extends FATServletClient {
      * some operations are performed to manually transfer them. Users should not do this. This test is only written
      * to experiment with what would happen and explore how to cope with it.
      */
-    @AllowedFFDC("org.apache.derby.client.am.XaException") // due to roll back & abort due to transaction timeout
     @Test
     public void testRemoveFailOverEnablementWhileServerIsRunning() throws Exception {
         // start with fail over enabled
@@ -632,14 +637,6 @@ public class SwitchFromSingleInstanceToFailOverTest extends FATServletClient {
      * some MBean operations are performed. Users should not do this. This test is only written to experiment with
      * what would happen and explore how to cope with it.
      */
-    @AllowedFFDC({
-        // due to transaction timeout:
-        "javax.transaction.RollbackException",
-        "javax.transaction.xa.XAException",
-        "javax.persistence.PersistenceException",
-        "java.lang.IllegalStateException",
-        "org.apache.derby.client.am.XaException"
-        })
     @Test
     public void testRemoveFailOverEnablementWhileServerIsStopped() throws Exception {
         // start with fail over enabled
@@ -744,10 +741,6 @@ public class SwitchFromSingleInstanceToFailOverTest extends FATServletClient {
      * While the server is running, removes the original instance, creating a new one with fail over enabled. Then verifies
      * that the previous, as well as new, tasks run.
      */
-    @AllowedFFDC({
-        "java.lang.IllegalStateException", // Attempting to execute an operation on a closed EntityManager // possible when task is still attempting to run after executor instance is removed
-        "org.apache.derby.client.am.XaException" // due to roll back & abort due to transaction timeout
-    })
     @Test
     public void testReplaceWithNewFailOverEnabledInstanceWhileServerIsRunning() throws Exception {
         ServerConfiguration config = originalConfig.clone();
@@ -839,7 +832,6 @@ public class SwitchFromSingleInstanceToFailOverTest extends FATServletClient {
      * Stops the server and removes the original instance, creating a new one with fail over enabled. Starts up the server and verifies
      * that the previous, as well as new, tasks run.
      */
-    @AllowedFFDC("org.apache.derby.client.am.XaException") // due to roll back & abort due to transaction timeout
     @Test
     public void testReplaceWithNewFailOverEnabledInstanceWhileServerIsStopped() throws Exception {
         ServerConfiguration config = originalConfig.clone();

